hub/src/main/java/ch/prevo/open/hub/nodes/NodeService.java

```diff
@@ -0,0 +1,68 @@
+package ch.prevo.open.hub.nodes;
+
+import ch.prevo.open.encrypted.model.InsurantInformation;
+import ch.prevo.open.encrypted.model.MatchNotification;
+import ch.prevo.open.hub.match.Match;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+
+@Service
+public class NodeService {
+
+    @Inject
+    private NodeRegistry nodeRegistry;
+
+    private final RestTemplate restTemplate;
+
+    public NodeService(RestTemplateBuilder restTemplateBuilder) {
+        this.restTemplate = restTemplateBuilder.build();
+    }
+
+    public Set<InsurantInformation> currentExits() {
+        Set<InsurantInformation> exits = new HashSet<>();
+        for (NodeConfiguration nodeConfig : nodeRegistry.currentNodes()) {
+            exits.addAll(lookupInsurantInformationList(nodeConfig.getJobExitsUrl()));
+        }
+        return exits;
+    }
+
+    public Set<InsurantInformation> currentEntries() {
+        Set<InsurantInformation> exits = new HashSet<>();
+        for (NodeConfiguration nodeConfig : nodeRegistry.currentNodes()) {
+            exits.addAll(lookupInsurantInformationList(nodeConfig.getJobEntriesUrl()));
+        }
+        return exits;
+    }
+
+    private List<InsurantInformation> lookupInsurantInformationList(String url) {
+        InsurantInformation[] nodeExits = restTemplate.getForObject(url, InsurantInformation[].class);
```

Comment:
What happens in the case of e.g. a 404 (or an other error). So if one node is not available it means that the matching is not able to proceed?

Reply:
This is currently not handled at all. I suggest we address the topic separately (error handling etc)